yarn-project/circuit-types/src/keys/key_store.ts

```diff
@@ -112,12 +96,25 @@ export interface KeyStore {
   getMasterIncomingViewingSecretKeyForPublicKey(masterIncomingViewingPublicKey: PublicKey): Promise<GrumpkinPrivateKey>;
 
   /**
-   * Retrieves public keys hash of the account
-   * @throws If the provided account address is not associated with any of the registered accounts.
-   * @param account - The account address to get public keys hash for.
-   * @returns A Promise that resolves to the public keys hash.
+   * Gets the key validation request for a given master public key hash and contract address.
+   * @throws If the account corresponding to the master public key hash does not exist in the key store.
+   * @param pkMHash - The master public key hash.
+   * @param contractAddress - The contract address to silo the secret key in the the key validation request with.
+   * @returns The key validation request.
    */
-  getPublicKeysHash(account: AztecAddress): Promise<Fr>;
+  getKeyValidationRequest(pkMHash: Fr, contractAddress: AztecAddress): Promise<KeyValidationRequest>;
 
+  /**
+   * Rotates the master nullifier key for the specified account.
+   *
+   * @dev This function updates the secret and public keys associated with the account.
+   * It appends a new secret key to the existing secret keys, derives the
+   * corresponding public key, and updates the stored keys accordingly.
+   *
+   * @param account - The account address for which the master nullifier key is being rotated.
+   * @param newSecretKey - (Optional) A new secret key of type Fq. If not provided, a random key is generated.
+   * @throws If the account does not have existing nullifier secret keys or public keys.
+   * @returns A Promise that resolves when the key rotation is complete.
+   */
   rotateMasterNullifierKey(account: AztecAddress, secretKey: Fq): Promise<void>;
```

Comment:
This function was missing TSDoc here so I copied it over from test_key_store.ts.